src/osd/PrimaryLogPG.cc

```diff
@@ -4143,6 +4143,21 @@ void PrimaryLogPG::log_op_stats(const OpRequest& op,
 	   << " inb " << inb
 	   << " outb " << outb
 	   << " lat " << latency << dendl;
+
+  if (m_dynamic_perf_stats.is_enabled()) {
```

Comment:
Nit: need locking

Reply:
I though the `pg` is locked at this stage (by `OSD::ShardedOpWQ::_process`)? Just retested with adding `assert(is_locked());` and it works.

Reply:
OK, so it's covered by the ```PG.lock()``` in OSD.cc (that was my concern).